maas/client/bones/tests/test_helpers.py

```diff
@@ -343,6 +357,39 @@ class TestAuthenticate(TestCase):
                 await helpers.authenticate(baseurl, "username", "password")
 
 
+class TestAuthenticateWithMacaroon(TestCase):
+
+    def setUp(self):
+        super().setUp()
+        self.mock_client_request = self.patch(Client, "request")
+        self.token_result = {
+            'consumer_key': 'abc', 'token_key': '123', 'token_secret': 'xyz'}
+        self.mock_response = Mock()
```

Comment:
You could use  real Response object here.

Reply:
It's kinda hard to do that, since you have to explicitly set attributes anyway, plus content is set via a private `response._content` attribute. I'd rather keep this since it's a bit more explicit.